azure-iot-device/samples/legacy-samples/register_symmetric_key.py

```diff
@@ -0,0 +1,68 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import logging
+from six.moves import input
+from azure.iot.device import SymmetricKeySecurityClient
+from azure.iot.device import create_from_security_client
+
+
+logging.basicConfig(level=logging.ERROR)
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+
+symmetric_key_security_client = SymmetricKeySecurityClient(registration_id, symmetric_key, id_scope)
+registration_client = create_from_security_client(
+    provisioning_host, symmetric_key_security_client, "mqtt"
+)
+
+
+def registration_status_callback(registration_result):
+    print("Operation Id =", registration_result.operation_id)
+    print("Status =", registration_result.status)
+    print(registration_result.status)
+    if registration_result.status == "assigned":
+        print("Device has been registered")
+    elif registration_result.status == "assigning":
```

Comment:
> elif registration_result.status == "assigning": [](start = 3, length = 48)

I'm not sure other SDKs call their callback for the intermediate step. I think they just wait until registration is successful or failed
I could see how this is useful but I wonder if we should make the default behavior to just call register and return when registration is complete. then if customers says they need callbacks for "progress tracking" we can always add it later. Given there's no "progress tracking" information i'm not sure it's useful anyway.

The async version would use await I guess?

Reply:
complete and return

---
In reply to: [277866057](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r277866057) [](ancestors = 277866057)